MetasysServices.Tests/MetasysClientTests.cs

```diff
@@ -2357,5 +2357,71 @@ public void TestMiscNullTokenValue()
         }
 
         #endregion
+
+        #region Ad-Hoc Calls
+        [Test]
+        public async Task TestSendAsyncWithAbsoluteUrl()
+        {
+            var requestUrl = "https://hostname/api/v5-preview/someController/someAction";
+            var httpRequest = new HttpRequestMessage(HttpMethod.Get, requestUrl);
+            await client.SendAsync(httpRequest);
+
+            httpTest.ShouldHaveCalled(requestUrl)
+                .WithVerb(HttpMethod.Get)
+                .Times(1);
+        }
+
+        [Test]
+        public async Task TestSendAsyncWithRelativeUrl()
+        {
+            var requestUrl = "api/v5-preview/someController/someAction";
+            var httpRequest = new HttpRequestMessage(HttpMethod.Get, requestUrl);
+            await client.SendAsync(httpRequest);
+
+            httpTest.ShouldHaveCalled($"https://hostname/{requestUrl}")
+                .WithVerb(HttpMethod.Get)
+                .Times(1);
+        }
+
+        [Test]
+        public async Task TestSendAsyncCheckQueryStringAndFragmentsAreUsed()
+        {
+            var requestUrl = $"api/v5-preview/someController/someAction?fqr=itemReference#fragment1";
+            var httpRequest = new HttpRequestMessage(HttpMethod.Get, requestUrl);
+            await client.SendAsync(httpRequest);
+
+            httpTest.ShouldHaveCalled($"https://hostname/{requestUrl}")
+                .WithVerb(HttpMethod.Get)
+                .WithQueryParamValue("fqr", "itemReference")
+                .With(call => call.Request.RequestUri.Fragment.StartsWith("#fragment1"))
+                .Times(1);
+        }
+
+        [Test]
+        public async Task TestSendAsyncCheckRequestHeadersAreUsed()
+        {
+            var requestUrl = "api/v5-preview/someController/someAction";
+            var httpRequest = new HttpRequestMessage(HttpMethod.Get, requestUrl);
+            httpRequest.Headers.Add("Accept", "application/json");
+            await client.SendAsync(httpRequest);
+
+            httpTest.ShouldHaveCalled($"https://hostname/{requestUrl}")
+                .WithVerb(HttpMethod.Get)
+                .WithHeader("Accept")
```

Comment:
How do you know this was with your header though? I would suggest using a more unique header value, and asserting the value too.

Reply:
Added custom header to assert. I guess there is bug in the `Flurl library` unable to assert the `header value`.
I am able to assert it with `header key` but with `header value` it throws `Flurl.Http.Testing.HttpCallAssertException`

Reply:
I think this failure is showing that our code is not correct!

I'm seeing that the `flurlRequest.WithHeaders` function is expecting an object or an `IDictionary`, yet you're providing an instance of `HttpRequestHeaders`, which is neither. So I'm guessing it's not properly converting to the right headers.

I did some debugging on my own and found that the flurl HTTP assertion was finding the header value to be `String[]`, which is clearly wrong!

I think this is all due to the clash between HTTP technically supporting multiple values for a header, but older versions of flurl didn't (and this repo currently is using an old version of flurl). See https://github.com/tmenier/Flurl/issues/541. I doubt we want to take on updating flurl for this PR, but this is something @Federico-Artioli needs to address.

Reply:
This code (from the version which basic-services currently references), shows that only 1 header value is supported by flurl and will try to stringify whatever the header value is. So that means we will need to write code accordingly, and warn users that only header values with 1 value is supported (which should fit most cases) until a future revision.

I can try and work on some changes to support this today. Ultimately, the test code here needs to check for the header value to ensure correct functionality.